src/start_env/mod.rs

```diff
@@ -326,16 +357,18 @@ fn start_btsieves(envfile_path: &PathBuf) -> impl Future<Item = Vec<Node<Btsieve
         })
 }
 
-fn start_cnds(envfile_path: &PathBuf) -> impl Future<Item = Vec<Node<Cnd>>, Error = Error> {
+fn start_cnds(
+    envfile_path: &PathBuf,
+    config_folder: PathBuf,
+    config_folder_str: String,
```

Comment:
:thinking: I know that `to_str` on a `PathBuf` returns an option, but is it really worse than passing in a redundant variable?

Reply:
You are welcome to have a go but it gets messy. That could go away with async/await I think.